VideoWeb/VideoWeb.UnitTests/Controllers/ConsultationController/StartConsultationTest.cs

```diff
@@ -236,5 +237,57 @@ public async Task Should_return_exception()
             var typedResult = (StatusCodeResult)result;
             typedResult.StatusCode.Should().Be((int)HttpStatusCode.InternalServerError);
         }
+
+        [TestCase(AppRoles.JudicialOfficeHolderRole, true)]
+        [TestCase(AppRoles.JudgeRole, true)]
+        [TestCase(AppRoles.StaffMember, true)]
+        [TestCase(AppRoles.CitizenRole, false)]
+        [TestCase(AppRoles.VhOfficerRole, false)]
+        [TestCase(AppRoles.VenueManagementRole, false)]
+        [TestCase(AppRoles.RepresentativeRole, false)]
+        [TestCase(AppRoles.CaseAdminRole, false)]
+        [TestCase(AppRoles.QuickLinkParticipant, false)]
+        [TestCase(AppRoles.QuickLinkObserver, false)]
+        public async Task When_a_user_tries_to_start_judge_joh_consultation_Should_return_correct_response_code_based_on_role(string role, bool shouldBeAllowed)
+        {
+            var controller = GetControllerWithContextForRole(role);
+            var request = ConsultationHelper.GetStartJohConsultationRequest(_testConference);
+            
+            var result =
+                await controller.StartConsultationAsync(
+                    ConsultationHelper.GetStartJohConsultationRequest(_testConference));
+            
+            // Assert
+            int timesCalledExpected;
+            if(shouldBeAllowed)
```

Comment:
can you make this two tests

Reply:
I could, but I'd just be duplicating the code before and after this statement. I feel like defining the role and expected result as test cases is a bit more preferable here.

Reply:
you could have had two tests one that checks when its allowed and one that doesnt to avoid using the logic here. You would still have test cases for the user roles I just meant moving the assertions into two different tests, ie,

` [TestCase(AppRoles.JudicialOfficeHolderRole)]
        [TestCase(AppRoles.JudgeRole)]
        [TestCase(AppRoles.StaffMember)]
        public async Task When_a_user_tries_to_start_judge_joh_consultation_Should_return_correct_response_code_based_on_role(string role)`
        
 `
        [TestCase(AppRoles.CitizenRole)]
        [TestCase(AppRoles.VhOfficerRole)]
        [TestCase(AppRoles.VenueManagementRole)]
        [TestCase(AppRoles.RepresentativeRole)]
        [TestCase(AppRoles.CaseAdminRole)]
        [TestCase(AppRoles.QuickLinkParticipant)]
        [TestCase(AppRoles.QuickLinkObserver)]
        public async Task When_a_user_tries_to_start_judge_joh_consultation_Should_return_correct_response_code_based_on_role(string role)`

Reply:
but its fine, its just a preference, normally I like avoiding use of if logics in unit tests